packages/appstore/src/components/modals/cashier-wallet-modal/cashier-wallet-modal.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Modal } from '@deriv/components';
+import { isMobile } from '@deriv/shared';
+import CashierWalletModalHeader from './cashier-wallet-modal-header';
+import CashierWalletModalBody from './cashier-wallet-modal-body';
+import { observer } from 'mobx-react-lite';
+import { useStore } from '@deriv/stores';
+
+const CashierWalletModal = observer(() => {
+    const store = useStore();
+    const {
+        client: { balance, currency },
+        ui: { is_dark_mode_on, is_cashier_wallet_modal_visible, setIsCashierWalletModalVisible },
+        traders_hub: { is_demo },
+    } = store;
+
+    const tab_content_ref = React.useRef<HTMLDivElement>(null);
+    const [active_tab_index, setActiveTabIndex] = React.useState(0);
+    const [show_wallet_name, setShowWalletName] = React.useState(true);
+
+    const closeModal = () => {
+        setIsCashierWalletModalVisible(false);
+        setActiveTabIndex(0);
+    };
+
+    const handleScroll = (e: Event) => {
+        const target = e.target as HTMLDivElement;
+        setShowWalletName(!(target.scrollTop > 0));
+    };
+
+    React.useEffect(() => {
+        const el_tab_content = tab_content_ref.current;
+        el_tab_content?.addEventListener('scroll', handleScroll);
+
+        return () => {
+            el_tab_content?.removeEventListener('scroll', handleScroll);
+        };
+    }, [active_tab_index, is_cashier_wallet_modal_visible]);
+
+    React.useEffect(() => {
+        if (isMobile()) {
+            setShowWalletName(true);
+        }
+    }, [active_tab_index, is_cashier_wallet_modal_visible]);
```

Comment:
@farzin-deriv believe that no. The current sense of of this useEffect is to reset the visibility of the wallet name to default state when we are switching between the tabs or closing the modal window. Since we should handle it only in mobile view, i added a check that if we are not in mobile view, we will not invoke setShowWalletName function.